neptune_lightgbm/impl/__init__.py

```diff
@@ -79,7 +94,79 @@ def __init__(self, run: 'neptune.Run', base_namespace=''):
 
     def __call__(self, env):
-        for name, loss_name, loss_value, _ in env.evaluation_result_list:
-            channel_name = '{}{}_{}'.format(self._base_namespace, name, loss_name)
+        # TODO:
+        # (data_name, eval_name, val, is_higher_better): `lightgbm.basic.Booster.__inner_eval`
+        # ('cv_agg', k, np.mean(v), metric_type[k], np.std(v)): `lightgbm.engine._agg_cv_result`
+        for name, loss_name, loss_value, *_ in env.evaluation_result_list:
+            channel_name = 'foo/{}/foo2/{}_{}'.format(self._base_namespace, name, loss_name)
             self._run[channel_name].log(loss_value, step=env.iteration)
 
+
+def create_booster_summary(
+        booster: Union[lgb.Booster, lgb.sklearn.LGBMModel],
+        log_importances: bool = True,
+        max_num_features: int = 10,
+        list_trees: list = None,
+        log_trees_as_dataframe: bool = True,  # works only for lgb.Booster
+        log_pickled_booster: bool = True,
+        log_trees: bool = False,  # requires graphviz
+        log_confusion_matrix: bool = False,  # requires scikit-plot
+        y_true: np.ndarray = None,
+        y_pred: np.ndarray = None,
+):
+    results_dict = {}
+    visuals_path = "visualizations/"
+    if log_importances:
+        split_plot = lgb.plot_importance(
+            booster,
+            importance_type="split",
+            title="Feature importance (split)",
+            max_num_features=max_num_features
+        )
+        gain_plot = lgb.plot_importance(
+            booster,
+            importance_type="gain",
+            title="Feature importance (gain)",
+            max_num_features=max_num_features
+        )
+        results_dict["{}feature_importances/split".format(visuals_path)] \
+            = neptune.types.File.as_image(split_plot.figure)
+        results_dict["{}feature_importances/gain".format(visuals_path)] \
+            = neptune.types.File.as_image(gain_plot.figure)
+
+    # TODO: Your file is larger than 15MB. Neptune supports logging files in-memory objects smaller than 15MB.
+    #       Resize or increase compression of this object
+
+    if log_trees:
+        try:
+            trees_series = []
+            for i in list_trees:
+                digraph = lgb.create_tree_digraph(booster, tree_index=i, show_info='data_percentage')
+                _, ax = plt.subplots(1, 1)
+                s = BytesIO()
+                s.write(digraph.pipe(format='png'))
+                s.seek(0)
+                ax.imshow(image.imread(s))
+                ax.axis('off')
+                trees_series.append(neptune.types.File.as_image(ax.figure))
+            results_dict["{}trees".format(visuals_path)] = neptune.types.FileSeries(trees_series)
+        except GraphvizExecutableNotFound:
+            # TODO: link to Neptune docs section where will be such info: https://graphviz.org/download/
+            log.warning(f"Trees won't be logged. To make it work install graphviz library on your OS.")
```

Comment:
python `graphviz` is default requirement, but if someone wants to use it, some C-librares are required in operation system.

* Do we  want to display this warning or be more restrictive and raise some kind of error?
* In this message should be link to neptune docs section about graphviz installation.

Reply:
UX should be similar to the XGBoost:
* Inform user that she needs to install additional libs
* warning, not error

https://github.com/neptune-ai/neptune-xgboost/blob/master/neptune_xgboost/impl/__init__.py#L54:L61

Reply:
Looks good, can you add "https://graphviz.org" to the warning. We will point user to the correct resource.